src/metabase/models/params/chain_filter.clj

```diff
@@ -200,7 +205,7 @@
                                            [:= :database.id database-id]
                                            [:not= :fk-field.fk_target_field_id nil]]})]
     (reduce
-     (partial merge-with merge)
+     (partial merge-with (partial merge-with into))
```

Comment:
We were losing info about joins here, since the format is `{src-table {dst-table [joins...]}}`, so the previous code never got to merge vectors. Strictly speaking it's not necessary, since `chain-filter-mbql-query` is using explicit join information, but given that it still was "broken" fundamentally (did not return it promised it would), I think it makes sense to keep it.